### PR TITLE
remove legacy auth cli actions that no longer work on current servers

### PR DESCRIPTION
**Pull request by deads2k**
We only support one version of skew for `oc` and `openshift-apiserver`, so the migration for RBAC and the policybindings no longer make sense.  This removes them

Intersection of @openshift/sig-security and @openshift/sig-master 

/assign @enj 
/assign @soltysh

### CODE REVIEW

File: pkg/oc/cli/util/clientcmd/gating.go [hunk 1]
```diff
@@ -1,126 +0,0 @@
...
-	if all {
-		return nil
-	}
-
-	return fmt.Errorf("the server does not support legacy policy resources")
-}
-
-// DiscoverGroupVersionResources performs a server resource discovery for each filterGVR, returning a slice of
-// GVRs for the matching Resources, and a bool for "all" indicating that each item in filterGVR was found.
-func DiscoverGroupVersionResources(client discovery.ServerResourcesInterface, filterGVR ...schema.GroupVersionResource) ([]schema.GroupVersionResource, bool, error) {
```
Comment (enj) [inline pkg/oc/cli/util/clientcmd/gating.go]: Hmm we may need this in the future?

File: pkg/oc/admin/migrate/authorization/authorization.go [hunk 2]
```diff
@@ -1,285 +0,0 @@
...
-	kerrs "k8s.io/apimachinery/pkg/api/errors"
-	"k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/client-go/util/flowcontrol"
-	rbacinternalversion "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/typed/rbac/internalversion"
-	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
-	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
-	"k8s.io/kubernetes/pkg/kubectl/resource"
-
-	authorizationapi "github.com/openshift/origin/pkg/authorization/apis/authorization"
-	"github.com/openshift/origin/pkg/authorization/registry/util"
```
Comment (enj) [inline pkg/oc/admin/migrate/authorization/authorization.go]: I think we can remove unused things in this package as well?

File: pkg/oc/admin/migrate/authorization/authorization.go [hunk 3]
```diff
@@ -1,285 +0,0 @@
...
-	// errOutOfSync is retriable since it could be caused by the controller lagging behind
-	errOutOfSync = migrate.ErrRetriable{errors.New("is not in sync with RBAC")}
-)
-
-type MigrateAuthorizationOptions struct {
-	migrate.ResourceOptions
-	rbac rbacinternalversion.RbacInterface
-}
-
-func NewCmdMigrateAuthorization(name, fullName string, f *clientcmd.Factory, in io.Reader, out, errout io.Writer) *cobra.Command {
```
Comment (enj) [inline pkg/oc/admin/migrate/authorization/authorization.go]: With this gone I believe `ResourceOptions.Complete` can be updated:

https://github.com/openshift/origin/blob/cd29885133474a180ef77ce596a9675f584920a1/pkg/oc/admin/migrate/migrator.go#L101-L108

File: pkg/oc/cli/util/clientcmd/gating.go (hunk 1 above)
Comment (deads2k) [reply pkg/oc/cli/util/clientcmd/gating.go]: > Hmm we may need this in the future?

And we can revisit at that time.  Git keeps the "might come in handy's" without us keeping cruft.

File: pkg/oc/admin/migrate/authorization/authorization.go (hunk 3 above)
Comment (deads2k) [reply pkg/oc/admin/migrate/authorization/authorization.go]: > With this gone I believe ResourceOptions.Complete can be updated:

ok

**Comment by deads2k** [comment]
> Not sure we want to remove the gating. @simo5 @mrogers950 any thoughts?

What are you using it for?  We don't keep code around because it might be handy some day.  For that you can go back in history or use the new primitives we'll have at that time. This is just debt to carry.  Found in this case because I'm making changes that force reconsideration of import.  Keeping it may seem free to you, but it costs others.

**Comment by deads2k** [comment]
updated.

File: pkg/oc/admin/migrate/migrator.go [hunk 4]
```diff
@@ -98,11 +98,10 @@ func (o *ResourceOptions) Bind(c *cobra.Command) {
 }
 
 func (o *ResourceOptions) Complete(f *clientcmd.Factory, c *cobra.Command) error {
-	// oc adm migrate authorization does not support printing and takes no parameters, so we ignore the error here
 	var flagErr error
```
Comment (enj) [inline pkg/oc/admin/migrate/migrator.go]: Remove the `flagErr` bits and just do `o.Output = kcmdutil.GetFlagString(c, "output")`.

**Pull request merged**